com.unity.render-pipelines.high-definition/Runtime/Lighting/PlanarReflectionFiltering.compute

```diff
@@ -0,0 +1,200 @@
+#pragma kernel FilterPlanarReflection
+#pragma kernel DownScale
+#pragma kernel DepthConversion
+
+#pragma only_renderers d3d11 playstation xboxone vulkan metal switch
+// #pragma enable_d3d11_debug_symbols
+
+// HDRP generic includes
+#include "Packages/com.unity.render-pipelines.core/ShaderLibrary/Common.hlsl"
+#include "Packages/com.unity.render-pipelines.high-definition/Runtime/ShaderLibrary/ShaderVariables.hlsl"
+#include "Packages/com.unity.render-pipelines.high-definition/Runtime/Material/Material.hlsl"
+
+// Tile size of this compute
+#define PLANAR_REFLECTION_TILE_SIZE 8
+
+// Mip chain of depth and color
+TEXTURE2D(_DepthTextureMipChain);
+TEXTURE2D(_ReflectionColorMipChain);
+
+CBUFFER_START(ShaderVariablesPlanarReflectionFiltering)
+    // The screen size (width, height, 1.0 / width, 1.0 / height) that is produced by the capture
+    float4 _CaptureBaseScreenSize;
+    // The screen size (width, height, 1.0 / width, 1.0 / height) of the current level processed
+    float4 _CaptureCurrentScreenSize;
+    // Normal of the planar reflection plane
+    float3 _ReflectionPlaneNormal;
+    // Roughness value of the current integration
+    float _IntegrationRoughness;
+    // World space position of the planar reflection (non camera relative)
+    float3 _ReflectionPlanePosition;
+    // FOV of the capture camera
+    float _CaptureCameraFOV;
+    // World space position of the capture camera (non camera relative)
+    float3 _CaptureCameraPositon;
+    // The mip index of the source data
+    int _SourceMipIndex;
+    // Inverse view projection of the capture camera (oblique)
+    float4x4 _CaptureCameraIVP;
+    // Inverse view projection of the capture camera (non oblique)
+    float4x4 _CaptureCameraIVP_NO;
+    // View projection of the capture camera (non oblique)
+    float4x4 _CaptureCameraVP_NO;
+    // Given that sometimes our writing texture can be bigger than the current target, we need to apply a scale factor before using the sampling intrinsic
+    float _RTScaleFactor;
+    // Far plane of the capture camera
+    float _CaptureCameraFarPlane;
+CBUFFER_END
+
+// Output buffer of our filtering code
+RW_TEXTURE2D(float4, _FilteredPlanarReflectionBuffer);
+
+// This function could be placed in a shared file.
+bool IntersectPlane(float3 rayOrigin, float3 rayDirection, float3 planePosition, float3 planeNormal, out float t)
```

Comment:
ifsn't this just plane / ray intersection, we ahve already a lot of them in GeometricTools.hlsl

Reply:
done